src/sentry/testutils/factories.py

```diff
@@ -2170,13 +2170,17 @@ def create_data_source(
     @assume_test_silo_mode(SiloMode.REGION)
     def create_detector(
         name: str | None = None,
+        config: dict | None = None,
         **kwargs,
     ) -> Detector:
         if name is None:
             name = petname.generate(2, " ", letters=10).title()
+        if config is None:
+            config = {}
```

Comment:
need to explicitly set this as a dictionary, the receiver does not think the Django `db_default={}` is the same as `{}`